magento/magento2
# Avoid binding magnifier events in duplicate on resize

### Description
A browser resize event will trigger the magnifier to be re-initialised. This includes binding a fresh set of event listeners to the image so that the magnifier will show as expected (on click or hover). However, there is no mechanism to unbind the old event listeners, so each time the browser triggers a resize event these will stack.
This is made worse by the 'hover' event type triggering the 'mouseover' event immediately after adding a listener for the same. This means that each resize event will consume progressively more CPU, without any additional value.

### Related Pull Requests
*None*

### Fixed Issues
*None*

### Manual testing scenarios
1. Visit a product page with magnifier in use and configured for hover events
1. Observe the resting CPU usage of the page
1. Resize the viewport (either height or width)
1. Observe a spike in CPU usage, which subsides
1. Resize the viewport several times
1. Observe a progressively longer spike in CPU usage
1. Continue resizing the viewport and notice that CPU eventually (after \~8 resizes in my experience) caps out at 100% for an unacceptable amount of time (several minutes), rendering the page useless.

<details><summary>Before & after pictures from JavaScript profiler</summary>

Before:
![Screenshot_2020-10-28_15-46-27](https://user-images.githubusercontent.com/334786/97470897-fd67bb80-193f-11eb-8106-1ce4c2b90db0.png)

After:
![Screenshot_2020-10-28_16-38-10](https://user-images.githubusercontent.com/334786/97470914-06588d00-1940-11eb-9228-0e3019cc66d1.png)

</details>

### Questions or comments
*None*

### Contribution checklist

### Resolved issues:
1. [x] resolves magento/magento2#30788: Avoid binding magnifier events in duplicate on resize
